README.md

```diff
@@ -429,3 +429,80 @@ Similarly, you can run the integration test suite via
 ```bash
 $ python run_tests.py tests/integration/
 ```
+
+
+i18n compatibility
+==================
+
+According to [edX docs on XBlock i18n][edx-docs-i18n], LMS runtime is capable of supporting XBlock i18n and l10n. 
+To comply with l10n requirements, XBlock is supposed to provide translations in 
+`xblock_package_root/translations/locale_code/LC_MESSAGES/text.po` folder in GNU Gettext Portable Object file format.
+
+[edx-docs-i18n]: http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/edx_platform/edx_lms.html#internationalization-support
+
+Drag and Drop v2 XBlock aims to comply with i18n requirements for Open edX platform, including a stricter set of 
+requirements for `edx.org` itself, thus providing the required files. So far only two translations are available:
+
+* Default English translation
+* Fake "Esperanto" translation used to test i18n/l10n.
+
+Updates to translated strings are supposed to be propagated to `text.po` files. Unfortunately, the process is mostly 
+manual. EdX [i18n_tools][edx-i18n-tools] might be helpful, but they are too tied to edx-platform, so as of Aug 2016 
+do not provide any benefits over manual approach.
+
+[edx-i18n-tools]: https://github.com/edx/i18n-tools
+
+In lieu of automated solution (i.e. edX i18n-tools supporting XBlock use case), one can use GNU Gettext toolkit
+to simplify manual updating `text.po` files.
+
+This command scrapes all the strings in all `*.py` files in `drag_and_drop_v2` folder and outputs `messages.po` file
+ in `drag_and_drop_v2` folder:
+
+```
+~/xblock-drag-and-drop-v2/drag_and_drop_v2$ find . -name "*.py" | xargs xgettext --language=python
+```
+
+Javascript command is a little bit more verbose:
+
+```
+~/xblock-drag-and-drop-v2/drag_and_drop_v2$ find . -name "*.js" -o  -path ./public/js/vendor -prune -a -type f | xargs xgettext --language=javascript --from-code=utf-8
+```
+
+Note that both commands generate partial `messages.po` file - JS or python only, while `test.po` is supposed to contain
+all translatable strings. Both commands can be modified to append to *exsisting* `messages.po` file by adding 
+`--join-existing` key.
+
+To check if `text.po` is correct, one can run `msgfmt` to build a `text.mo` file:
+
+```
+~/xblock-drag-and-drop-v2/drag_and_drop_v2$ msgfmt translations/en/LC_MESSAGES/text.po -o translations/en/LC_MESSAGES/text.mo 
+```
+
+If everything is correct, it will silently exit and create `translations/en/LC_MESSAGES/text.mo` file.
+
+**Note:** As of 2016-08-04, XBlock translation feature was (partially?) broken, so switching to other locale did not 
+cause Drag and Drop v2 to be fully translated - only some strings that happen to exist in edx-platform translation
+files are translated. This is a bug on edx-platform side.
```

Comment:
I don't believe this is accurate. As far as I know, the only issue is that JavaScript i18n for XBlocks has not yet been implemented (see [here](https://openedx.atlassian.net/browse/WL-414)) - the JS `gettext()` call only loads strings from the platform, not from the current XBlock. In other words, this is just a missing/future feature, not a bug.

However, if something other than JS `gettext()` is currently not working, that is a bug and should be reported elsewhere so that it can get fixed.